util/mk1mf.pl

```diff
@@ -304,6 +304,7 @@
 $cflags.=" -DOPENSSL_NO_ASYNC" if $no_async;
 $cflags.=" -DOPENSSL_NO_AUTOALGINIT" if $no_autoalginit;
 $cflags.=" -DOPENSSL_NO_AUTOERRINIT" if $no_autoerrinit;
+$cflags.=" -DOPENSSL_NO_NEXTPROTONEG" if $no_nextprotoneg;
```

Comment:
You are both correct.  Toss the `$cflags.=...` line, but add this in `%valid_options`:

     		"no-nextprotoneg" => 0,